vm/vm.go

```diff
@@ -1208,3 +1224,21 @@ func (vm *VM) Fatal(msg string, fields ...zap.Field) {
 	vm.snowCtx.Log.Fatal(msg, fields...)
 	panic("fatal error")
 }
+
+type apiCoreContext struct {
+	*VM
+}
```

Comment:
Could we remove `apiCoreContext` and add `GetDataDir` and `GetGenesisBytes` directly to the VM in `resolutions.go` ? Not sure why the intermediate type is needed